File: LeetCode/Medium/Count Good Node In Binary Tree/Java/countGoodNodes.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int good;
    public int goodNodes(TreeNode root) {
        good = 1;
        _goodNodes(root, root.val);
        return good;
    }

    private void _goodNodes(TreeNode root, int val) {
        if (root == null) return;
        if (root.left != null && root.left.val >= val) {
            good++;
            _goodNodes(root.left, root.left.val);
        } else {
            _goodNodes(root.left, val);
        }

        if (root.right != null && root.right.val >= val) {
            good++;
            _goodNodes(root.right, root.right.val);
        } else {
            _goodNodes(root.right, val);
        }
    }
}

File: LeetCode/Medium/Count Good Node In Binary Tree/C/count_good_nodes.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */

void recur(struct TreeNode* root, int val, int* good)
{
    if (root == NULL) return;
    if (root->left != NULL && val <= root->left->val)
    {
        (*good)++;
        recur(root->left, root->left->val, good);
    }
    else
    {
        recur(root->left, val, good);
    }

    if (root->right != NULL && val <= root->right->val)
    {
        (*good)++;
        recur(root->right, root->right->val, good);
    }
    else
    {
        recur(root->right, val, good);
    }
}

int goodNodes(struct TreeNode* root)
{
    int good = 1;
    recur(root, root->val, &good);
    return good;
}
